src/services/completions.ts

```diff
@@ -5384,7 +5385,7 @@ function getApparentProperties(type: Type, node: ObjectLiteralExpression | JsxAt
 }
 
 function containsNonPublicProperties(props: Symbol[]) {
-    return some(props, p => !!(getDeclarationModifierFlagsFromSymbol(p) & ModifierFlags.NonPublicAccessibilityModifier));
+    return some(props, p => !!(getDeclarationModifierFlagsFromSymbol(p) & ModifierFlags.NonPublicAccessibilityModifier) || !!p.valueDeclaration && isNamedDeclaration(p.valueDeclaration) && isPrivateIdentifier(p.valueDeclaration.name));

```

Comment:
likely this alone (or something close to it) would fix the completion problem (`private foo` is already filtered out from completions, only `#foo` is not) but that wouldn't fix the `contextuallyTypedParametersPositionIncludesClassWithPrivateMember` test cases